Annotations refactoring & PHP 8 Attributes (#801)
overblog/GraphQLBundle: docs/annotations/annotations-reference.md

}
```

## Annotations shortcuts

When using annotations or attributes, you can use a shortened version of the syntax.  
Example without:  
`@GQL\Field(name="MyField")` 

> Exampel with shortcut:  

Reviewer: ```suggestion
Example with shortcut:  
```

`@GQL\Field("MyField")`

> When using the shortened version, it is the attribute describe as `default attribute` in the following list that will be set if attribute name is not specified.  

Reviewer: This sentence is built wrong. I would fix it, but I just don't understand it 😅

## @Arg

> This annotation is used in conjonction with a `@Field`, a `@Query` or `@Mutation` to define an argument.

Reviewer: ```suggestion
This annotation is used in conjunction with a `@Field`, a `@Query` or `@Mutation` to define an argument.
```

I feel a French trace here 😏

 */
class Hero {
    /**
     * @GQL\Field
     * @GQL\FieldBuilder("GenericIdBuilder", config={"name": "heroId"})
     */
    public $id;

    /**
     * @GQL\Field(type="[Hero]", resolve="resolver('hero_friends', [args['droidsOnly'], args['nameStartsWith']])")
     * @GQL\Arg(name="droidsOnly", type="Boolean", description="Retrieve only droids heroes"),
     * @GQL\Arg(name="nameStartsWith", type="String", description="Retrieve only heroes with name starting with")
     */
    public $friends;
}
```

## @ArgsBuilder

> This annotation is used in conjonction with a `@Field`, a `@Query` or `@Mutation` to generate the field arguments.
> It is used to set a arguments builder for a field (see [Args builders](../definitions/builders/args.md)))

Reviewer: ```suggestion
This annotation is used in conjunction with a `@Field`, a `@Query` or `@Mutation` to generate the field arguments.
It is used to set an arguments builder for a field (see [Args builders](../definitions/builders/args.md)))
```
